grpclog/component.go

```diff
@@ -0,0 +1,162 @@
+var prefixVars = map[string]*componentData{}
+var cache = map[string]*componentData{}
+
+// apply applies the parameter componentData to the receiver componentData.
+// Sentinel values in the parameter will not be applied.
+func (c *componentData) apply(applyData *componentData) {
+	if applyData.verbosity != sentinel {
+		c.verbosity = applyData.verbosity
+	}
+	if applyData.level != sentinel {
+		c.level = applyData.level
+	}
+}
+
+func (c *componentData) InfoDepth(depth int, args ...interface{}) {
+	if c.level > levelInfo {
+		return
+	}
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.InfoDepth(depth, args...)
+}
+
+func (c *componentData) WarningDepth(depth int, args ...interface{}) {
+	if c.level > levelWarning {
+		return
+	}
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.WarningDepth(depth, args...)
+}
+
+func (c *componentData) ErrorDepth(depth int, args ...interface{}) {
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.ErrorDepth(depth, args...)
+}
+
+func (c *componentData) FatalDepth(depth int, args ...interface{}) {
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.FatalDepth(depth, args...)
+}
+
+func (c *componentData) Info(args ...interface{}) {
+	c.InfoDepth(0, args...)
+}
+
+func (c *componentData) Warning(args ...interface{}) {
+	c.WarningDepth(0, args...)
+}
+
+func (c *componentData) Error(args ...interface{}) {
+	c.ErrorDepth(0, args...)
+}
+
+func (c *componentData) Fatal(args ...interface{}) {
+	c.FatalDepth(0, args...)
+}
+
+func (c *componentData) Infof(format string, args ...interface{}) {
+	c.InfoDepth(0, fmt.Sprintf(format, args...))
+}
+
+func (c *componentData) Warningf(format string, args ...interface{}) {
+	c.WarningDepth(0, fmt.Sprintf(format, args...))
+}
+
+func (c *componentData) Errorf(format string, args ...interface{}) {
+	c.ErrorDepth(0, fmt.Sprintf(format, args...))
+}
+
+func (c *componentData) Fatalf(format string, args ...interface{}) {
+	c.FatalDepth(0, fmt.Sprintf(format, args...))
+}
+
+func (c *componentData) Infoln(args ...interface{}) {
+	c.Info(args...)
+}
+
+func (c *componentData) Warningln(args ...interface{}) {
+	c.Warning(args...)
+}
+
+func (c *componentData) Errorln(args ...interface{}) {
+	c.Error(args...)
+}
+
+func (c *componentData) Fatalln(args ...interface{}) {
+	c.Fatal(args...)
+}
+
+func (c *componentData) V(l int) bool {
+	if c.verbosity == sentinel {
+		return grpclog.Logger.V(l)
+	}
+	return c.verbosity >= l
+}
+
+// Component creates a new component and returns it for logging. If a component
+// with the name already exists, nothing will be created and it will be
+// returned. SetLoggerV2 will panic if it is called with a logger created by
+// Component.
+func Component(componentName string) DepthLoggerV2 {
+	if cData, ok := cache[componentName]; ok {
+		return cData
+	}
+	c := &componentData{componentName, sentinel, levelInfo}
+	// Apply prefix settings
```

Comment:
Delete applying settings. This function just needs to `return &componentData{name: componentName}`
(It needs the logic to check cache, so keep that).

Reply:
Updated this method.